File: solutions/cpp/resistor-color-duo/resistor_color_duo.cpp
#include "resistor_color_duo.h"

#include <algorithm>

namespace resistor_color_duo
{
    using namespace std;

    // Better Be Right Or Your Great Big Values Go Wrong
    const vector<string> color_codes{
        "black", "brown", "red", "orange", "yellow",
        "green", "blue", "violet", "grey", "white"};

    int color_code(string color)
    {
        auto it = find(color_codes.begin(), color_codes.end(), color);
        return distance(color_codes.begin(), it);
    }

    int value(vector<string> colors)
    {
        return color_code(colors[0]) * 10 + color_code(colors[1]);
    }
} // namespace resistor_color_duo

File: solutions/lua/resistor-color-duo/resistor-color-duo.lua
-- Better Be Right Or Your Great Big Values Go Wrong
local colors = {
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white"
}

-- Return the value of a color band
local color_code = function(color)
  for index, value in ipairs(colors) do
    if value == color then
      return index - 1
    end
  end
end

return {
  -- Calculate a resistance value from two colors
  value = function(colors)
    return color_code(colors[1]) * 10 + color_code(colors[2])
  end
}
